react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+
+class ValidateInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      entry: 'start',
+      userPassword: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(event) {
+    this.setState({
+      userPassword: event.target.value
+    });
+    event.preventDefault();
+  }
+
+  handleChange(event) {
+    if (event.target.value.length >= 8) {
+      this.setState({
+        entry: 'pass'
+      });
+    } else if (event.target.value.length === 0) {
+      this.setState({
+        entry: 'empty'
+      });
+    } else if (event.target.value.length < 8) {
+      this.setState({
+        entry: 'short'
+      });
+    }
```

Comment:
I recommend also setting the state of the `userPassword` at the same time you set the validation status.